libs/application-generic/src/usecases/create-workflow/create-workflow.usecase.ts

```diff
@@ -136,6 +120,29 @@ export class CreateWorkflow {
     return storedWorkflow;
   }
 
+  private generateTriggerIdentifier(command: CreateWorkflowCommand) {
+    if (command.triggerIdentifier) {
+      return command.triggerIdentifier;
+    }
+
+    let triggerIdentifier: string;
+    if (command.type === WorkflowTypeEnum.BRIDGE)
+      /*
+       * Bridge workflows need to have the identifier preserved to ensure that
```

Comment:
In code-first, we allow human-friendly workflow identifiers. This contradicts the latest definition of a workflow identifier as defined in the Dashboard as workflowIds defined in code are not necessarily slugs. They can contain spaces, tabs, and uppercase letters. 

My suggestion is that with need to be stricter in @novu/framework when it comes to workflowIds in TS and apply the same rules that will be applied to workflowIds degined in Dashboard.

Reply:
If i am not wrong the code here will stay the same, i will create a ticket about this topic in order to update framework.